sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -499,6 +503,11 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
       StructType(schema.filterNot(_.name == parsedOptions.columnNameOfCorruptRecord))
 
     val linesWithoutHeader: RDD[String] = maybeFirstLine.map { firstLine =>
+      if (!parsedOptions.enforceSchema) {
```

Comment:
What does this mean? It sounds like we already decide to issue a warning log message even if `enforceSchema` is true.

Reply:
fixed